qa/os/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -0,0 +1,224 @@
+import org.elasticsearch.packaging.util.Shell.Result;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static java.nio.file.attribute.PosixFilePermissions.fromString;
+import static org.elasticsearch.packaging.util.Docker.assertPermissionsAndOwnership;
+import static org.elasticsearch.packaging.util.Docker.copyFromContainer;
+import static org.elasticsearch.packaging.util.Docker.ensureImageIsLoaded;
+import static org.elasticsearch.packaging.util.Docker.existsInContainer;
+import static org.elasticsearch.packaging.util.Docker.removeContainer;
+import static org.elasticsearch.packaging.util.Docker.runContainer;
+import static org.elasticsearch.packaging.util.Docker.verifyContainerInstallation;
+import static org.elasticsearch.packaging.util.Docker.waitForPathToExist;
+import static org.elasticsearch.packaging.util.FileMatcher.p660;
+import static org.elasticsearch.packaging.util.FileUtils.append;
+import static org.elasticsearch.packaging.util.FileUtils.getTempDir;
+import static org.elasticsearch.packaging.util.FileUtils.mkdir;
+import static org.elasticsearch.packaging.util.FileUtils.rm;
+import static org.elasticsearch.packaging.util.ServerUtils.makeRequest;
+import static org.elasticsearch.packaging.util.ServerUtils.waitForElasticsearch;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.emptyString;
+import static org.junit.Assume.assumeTrue;
+
+public class DockerTests extends PackagingTestCase {
+    protected DockerShell sh;
+
+    @BeforeClass
+    public static void filterDistros() {
+        assumeTrue("only Docker", distribution.isDocker());
+
+        ensureImageIsLoaded(distribution);
+    }
+
+    @AfterClass
+    public static void cleanup() {
+        // runContainer also calls this, so we don't need this method to be annotated as `@After`
+        removeContainer();
+    }
+
+    @Before
+    public void setupTest() throws Exception {
+        sh = new DockerShell();
+        installation = runContainer(distribution());
+    }
+
+    /**
+     * Checks that the Docker image can be run, and that it passes various checks.
+     */
+    public void test10Install() {
+        verifyContainerInstallation(installation, distribution());
+    }
+
+    /**
+     * Checks that no plugins are initially active.
+     */
+    public void test20PluginsListWithNoPlugins() {
+        final Installation.Executables bin = installation.executables();
+        final Result r = sh.run(bin.elasticsearchPlugin + " list");
+
+        assertThat("Expected no plugins to be listed", r.stdout, emptyString());
+    }
+
+    /**
+     * Check that a keystore can be manually created using the provided CLI tool.
+     */
+    public void test40CreateKeystoreManually() throws InterruptedException {
```

Comment:
This doesn't really match with the pattern we've been working towards. We want test classes for categories of checks, like keystore related (which @williamrandolph has been splitting out in a separate PR). Which distribution is being used should not matter to the test, only that we know how to do primitive operations for the given distribution (eg start, stop, run cli tool, etc)